plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-related-products-controller.php

```diff
@@ -0,0 +1,181 @@
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	public function get_collection_params() {
+		$params                       = array();
+		$params['context']            = $this->get_context_param();
+		$params['context']['default'] = 'view';
+
+		$params['categories'] = array(
+			'description'       => __( 'Limit result set to specific product categorie ids.', 'woocommerce' ),
+			'type'              => 'array',
+			'items'             => array(
+				'type' => 'integer',
+			),
+			'default'           => array(),
+			'sanitize_callback' => 'wp_parse_id_list',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['tags']       = array(
+			'description'       => __( 'Limit result set to specific product tag ids.', 'woocommerce' ),
+			'type'              => 'array',
+			'items'             => array(
+				'type' => 'integer',
+			),
+			'default'           => array(),
+			'validate_callback' => 'rest_validate_request_arg',
+			'sanitize_callback' => 'wp_parse_id_list',
+		);
+
+		return $params;
+	}
+
+	/**
+	 * Get the related product categories.
+	 */
+	public function get_related_product_cat_terms( $terms, $id ) {
+		return $this->categories;
+	}
+
+	/**
+	 * Get the related product tags.
+	 */
+	public function get_related_product_tag_terms( $terms, $id ) {
+		return $this->tags;
+	}
+
+	/**
+	 * Get the related products.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return object
+	 */
+	public function get_items( $request ) {
+		$id     = $request->get_param( 'id' );
+		$object = wc_get_product( $id );
+
+		$categories = $request->get_param( 'categories' );
+		$tags       = $request->get_param( 'tags' );
+
+		/*
+		 * If categories param is defined,
+		 * filter the related products by the categories.
+		 */
+		if ( ! empty( $categories ) ) {
+			$this->categories = $categories;
+			add_filter(
+				'woocommerce_get_related_product_cat_terms',
+				array( $this, 'get_related_product_cat_terms' ),
+				100, 2
+			);
+		}
+
+		/*
+		 * If tags param is defined,
+		 * filter the related products by the tags.
+		 */
+		if ( ! empty( $tags ) ) {
+			$this->tags = $tags;
+			add_filter(
+				'woocommerce_get_related_product_tag_terms',
+				array( $this, 'get_related_product_tag_terms' ),
+				100, 2
+			);
+		}
+
+		$ids = wc_get_related_products( $id );
+
+		// Remove the product categories filter.
+		remove_filter( 'woocommerce_get_related_product_cat_terms', array( $this, 'get_related_product_cat_terms' ), 100, 2 );
+
+		// Remove the product tags filter.
+		remove_filter( 'woocommerce_get_related_product_tag_terms', array( $this, 'get_related_product_tag_terms' ), 100, 2 );
+
+		return (object) array(
```

Comment:
I was expecting this endpoint to respond with a list of products not only their ids?
Similar to what we have for variations. Not sure on this. What do you think @louwie17 ?

Reply:
Yes, I agree that we should return a list of products similar to the [`/products` endpoint](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products).

It may be good to add support for the `_fields` param if that is easy to do, that way if you do only want `id's` a user can do `_fields: ['id']`.
Maybe could make use of this function already: https://github.com/woocommerce/woocommerce/blob/12940ebcd60bf72857f8b080dbccb47638b74e7c/plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-controller.php#L529
It may get auto added if you extend from `WC_REST_Posts_Controller`